tests/rename/24.t

```diff
@@ -18,17 +18,38 @@ expect 0 mkdir ${dst_parent} 0755
 expect 0 mkdir ${src_parent}/${src} 0755
 cdir=`pwd`
 
+# Detect filesystem type
+fs_type=$(stat -f -c %T . 2>/dev/null || echo unknown)
+
 # Initial conditions
-expect 3 lstat ${src_parent} nlink
-expect 2 lstat ${dst_parent} nlink
+if [ "$fs_type" = "btrfs" ]; then
+	todo Linux "Btrfs uses CoW; link count semantics differ from POSIX."
+	expect 3 lstat ${src_parent} nlink
+
+	todo Linux "Btrfs uses CoW; link count semantics differ from POSIX."
+	expect 2 lstat ${dst_parent} nlink
+else
+	expect 3 lstat ${src_parent} nlink
+	expect 2 lstat ${dst_parent} nlink
+fi
+
 dotdot_inode=`${fstest} lstat ${src_parent} inode`
 expect ${dotdot_inode} lstat ${src_parent}/${src}/.. inode
 
 expect 0 rename ${src_parent}/${src} ${dst_parent}/${dst}
 
 # The .. link and parents' nlinks values should be updated
-expect 2 lstat ${src_parent} nlink
-expect 3 lstat ${dst_parent} nlink
+if [ "$fs_type" = "btrfs" ]; then
```

Comment:
Ditto: above
```suggestion
if [ "$fs" = "btrfs" ]; then
```

Reply:
Odd. That should be defined. Is `conf` present and being sourced properly?

Reply:
You were right - $fs is properly defined from the conf file. Earlier it seemed undefined because my check was comparing against lowercase btrfs, while the variable was actually set to uppercase BTRFS.
I’ve updated the condition to handle the correct value and $fs now works as expected. The test passes on BTRFS as well after this change.